src/source-control/GitSourceControl.ts

```diff
@@ -206,24 +191,17 @@ export class GitSourceControl implements SourceControl {
       earliestInterestingCommit,
       commits: commitHashMap,
     };
-
     this.repo = ourRepository;
   }
 
   async getCommitByHash(hash: CommitHash): Promise<Commit | null> {
-    // TODO: Implement for partial match
-    const relatedCommits: Commit[] = [];
-    const keys = Array.from(this.repo.commits.keys());
-    keys.forEach((key) => {
-      if (key.indexOf(hash) !== -1) {
-        // If hash in key, as the key was from the commits' key, it should be present
-        relatedCommits.push(this.repo.commits.get(key)!);
-      }
-    });
-    if (relatedCommits.length !== 1) {
+    try {
+      const repo = await nodegit.Repository.open(this.repoPath);
+      const commit = await nodegit.AnnotatedCommit.fromRevspec(repo, hash);
+      const completeCommitHash = commit.id().tostrS();
+      return this.repo.commits.get(completeCommitHash) || null;
+    } catch (err) {
```

Comment:
Could we log all our caught errors? Failing silently usually makes debugging really difficult.

If we don't expect any errors to be thrown we should also not catch them at all, but iirc yesterday you mentioned that something here will throw, so I guess logging it is the best solution.

I think it'll be even nicer if we caught all errors in Stacker, but let's figure that out another time.

Reply:
Would this work instead? `throw new Error(err.message);`

Reply:
The most ideal way will be to define custom error classes and throw those, but I think for now we can just log them and return null